pallets/account-linker/src/tests.rs

```diff
@@ -45,20 +78,134 @@ fn it_works_for_default_value() {
+		assert_eq!(result.err(), Some(DispatchError::from(AccountLinkerError::LinkRequestExpired)));
+	});
+}
+
+#[test]
+fn test_insert_eth_address() {
+	new_test_ext().execute_with(|| {
+
+		let account: u64 = 5;
+		let block_number: u64 = 99999;
+
+
+		let mut gen = Random{};
+		for i in 0..(MAX_ETH_LINKS) {
+
+			let key_pair = gen.generate().unwrap();
+
+			let msg = generate_msg(account, block_number + i as u64);
+			let sig = generate_sig(&key_pair, &msg);
+
+			let (r, s, v) = generate_rsv(&sig);
+
+			let _ = AccountLinker::link(Origin::signed(1),
+										account,
+										i as u32,
+										block_number + i as u64,
+										r,
+										s,
+										v);
+			assert_eq!(AccountLinker::eth_addresses(account).len(), i+1);
+
+			let inserted_addr = AccountLinker::eth_addresses(account)[i];
+			let recovered_addr = util::addr_from_sig(msg.into(), sig).ok().unwrap();
+			assert_eq!(inserted_addr, recovered_addr);
+		}
+	});
+}
+
+#[test]
+fn test_update_eth_address() {
+	new_test_ext().execute_with(|| {
+
+		let account: u64 = 5;
+		let block_number: u64 = 99999;
+
+		let mut gen = Random{};
+		for i in 0..(MAX_ETH_LINKS) {
+			let key_pair = gen.generate().unwrap();
+			let msg = generate_msg(account, block_number + i as u64);
+			let sig = generate_sig(&key_pair, &msg);
+			let (r, s, v) = generate_rsv(&sig);
+
+			let _ = AccountLinker::link(Origin::signed(1),
+										account,
+										i as u32,
+										block_number + i as u64,
+										r,
+										s,
+										v);
+		}
+
+		let index: u32 = 2 as u32;
+		// Retrieve previous addr
+		let addr_before_update =  AccountLinker::eth_addresses(&account)[index as usize];
+		// Update addr at slot `index`
+		let key_pair = gen.generate().unwrap();
+
+		let msg = generate_msg(account, block_number + 9 as u64);
+		let sig = generate_sig(&key_pair, &msg);
+		let (r, s, v) = generate_rsv(&sig);
+
+		let _ = AccountLinker::link(Origin::signed(1),
+									account,
+									index,
+									block_number + 9 as u64,
+									r,
+									s,
+									v);
+		let updated_addr =  AccountLinker::eth_addresses(&account)[index as usize];
+		let recovered_addr = util::addr_from_sig(msg.into(), sig).ok().unwrap();
+		assert_ne!(updated_addr, addr_before_update);
+		assert_eq!(updated_addr, recovered_addr);
+	});
+}
+
+
+
+#[test]
+fn test_eth_address_pool_overflow() {
+	new_test_ext().execute_with(|| {
+
+		let account: u64 = 5;
+		let block_number: u64 = 99999;
+
+		let mut gen = Random{};
+		let key_pair = gen.generate().unwrap();
+
+		let msg = generate_msg(account, block_number);
+		let sig = generate_sig(&key_pair, &msg);
+		let (r, s, v) = generate_rsv(&sig);
+
+		let index: u32 = (MAX_ETH_LINKS * 2) as u32;
+		for _ in 0..(MAX_ETH_LINKS+1) {
+			let _ = AccountLinker::link(Origin::signed(1),
+										account,
+										index,
+										block_number,
+										r,
+										s,
+										v);
+		}
+		assert_eq!(AccountLinker::eth_addresses(account).len(), MAX_ETH_LINKS);
```

Comment:
ditto (compare the whole vec)

Reply:
No need to do this, I only want to check overflow.
As for the content, I checked them in previous cases (insert and update).